src/bootupd.rs

```diff
@@ -680,25 +633,103 @@ pub(crate) fn client_run_migrate_static_grub_config() -> Result<()> {
             println!("GRUB config symlink successfully replaced with the current config");
 
             // Remove the now unused symlink (optional cleanup, ignore any failures)
-            _ = dirfd.remove_file(&stripped_config);
+            _ = dirfd.remove_file(stripped_config);
         }
     };
 
     println!("Setting 'sysroot.bootloader' to 'none' in ostree repo config...");
+    set_ostree_bootloader("none")?;
+
+    println!("Static GRUB config migration completed successfully");
+    Ok(())
+}
+
+#[context("Get sysroot.bootloader in ostree repo config")]
+pub(crate) fn get_ostree_bootloader() -> Result<Option<String>> {
+    let mut cmd = std::process::Command::new("ostree");
+    let result = cmd
+        .args([
+            "config",
+            "--repo=/sysroot/ostree/repo",
+            "get",
+            "sysroot.bootloader",
+        ])
+        .output()
+        .context("Querying ostree sysroot.bootloader")?;
+    if !result.status.success() {
+        // ostree will exit with a non zero return code if the key does not exists
+        println!("ostree repo 'sysroot.bootloader' config option is not set yet");
```

Comment:
Should this print statement move to the caller? Feels odd to have a helper print things, but also it then makes it consistent with the other branches in that call on the callers side.

Reply:
SGTM, will move it to caller.